feat: Implement Actions property for ComputerSystem endpoint

amarnath-ac opened the pull request:
- Added Actions schema to OpenAPI computer system spec file
- Implement Actions property for ComputerSystem endpoint
- Enhanced unit tests with Actions validation

Computer systems output:
curl -k -u "standalone:G@ppm0ym" http://localhost:8181/redfish/v1/Systems/a72605c3-7494-4c63-855b-273410b4e163 | jq .
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   732  100   732    0     0    150      0  0:00:04  0:00:04 --:--:--   199
{
  "@odata.context": "/redfish/v1/$metadata#ComputerSystem.ComputerSystem",
  "@odata.id": "/redfish/v1/Systems/a72605c3-7494-4c63-855b-273410b4e163",
  "@odata.type": "#ComputerSystem.v1_26_0.ComputerSystem",
  "Actions": {
    "#ComputerSystem.Reset": {
      "target": "/redfish/v1/Systems/a72605c3-7494-4c63-855b-273410b4e163/Actions/ComputerSystem.Reset",
      "title": "Reset"
    }
  },
  "HostName": null,
  "Id": "a72605c3-7494-4c63-855b-273410b4e163",
  "Manufacturer": "Intel(R) Client Systems",
  "Model": "NUC12WSKv5",
  "Name": "a72605c3-7494-4c63-855b-273410b4e163",
  "PowerState": "On",
  "SerialNumber": "BTWS22800956",
  "SystemType": "Physical"
}

amarnath-ac committed:
feat: Implement Actions property for ComputerSystem endpoint

- Added Actions schema to OpenAPI computer system spec file
- Implement Actions property for ComputerSystem endpoint
- Enhanced unit tests with Actions validation

Signed-off-by: C, Amarnath <amarnath.c@intel.com>

sudhir-intc commented on redfish/internal/usecase/computer_system.go:
```diff
@@ -323,3 +327,35 @@ func (uc *ComputerSystemUseCase) convertStateToGenerated(state string) *generate
+	title := "Reset"
+
+	// Define the allowable reset types
+	allowableResetTypes := []generated.ResourceResetType{
```
check the content to what we have implemented - see this one : https://github.com/nmgaston/console/blob/35ace115d03b084ee25382df245d51d347756c49/redfish/internal/usecase/wsman_repo.go#L464 
We support only the following:
case redfishv1.PowerStateOn:
		return devices.CIMPMSPowerOn, nil // Power On = 2
	case redfishv1.PowerStateOff:
		return powerActionPowerDown, nil
	case redfishv1.ResetTypeForceOff:
		return powerActionPowerDown, nil
	case redfishv1.ResetTypeForceRestart:
		return powerActionReset, nil
	case redfishv1.ResetTypePowerCycle:
		return powerActionPowerCycle, nil

amarnath-ac replied:
Allowable types are removed because spec does not support it. To be 100% spec compliance I recommend not to add allowable reset types in actions reset. 
If we decide to support allowable reset types in actions reset then 2 approaches: 
1. update spec with allowable reset types in actions reset by taking exception (My recommendation so that its clean implementation)
2. Manually update in code

sudhir-intc commented on redfish/openapi/dmtf/ComputerSystem.v1_26_0.yaml:
```diff
@@ -1,5 +1,80 @@
+      x-patternProperties:
+        ^([a-zA-Z_][a-zA-Z0-9_]*)?@(odata|Redfish|Message)\.[a-zA-Z_][a-zA-Z0-9_]*$:
+          description: This property shall specify a valid odata or Redfish property.
+    ComputerSystem_v1_26_0_OemActions:
```
OemActions would not be needed since we are not support OemActions for now

amarnath-ac replied:
removed

sudhir-intc commented on redfish/openapi/dmtf/ComputerSystem.v1_26_0.yaml:
```diff
@@ -1,5 +1,80 @@
+      x-patternProperties:
+        ^([a-zA-Z_][a-zA-Z0-9_]*)?@(odata|Redfish|Message)\.[a-zA-Z_][a-zA-Z0-9_]*$:
+          description: This property shall specify a valid odata or Redfish property.
+    ComputerSystem_v1_26_0_Reset:
```
Please ensure the schemas maintain the same sorted order as the original spec. This will help us to compare what was added/modified

amarnath-ac replied:
done

amarnath-ac committed:
refactor: Update Actions structure to comply with DMTF Redfish specification

- Update Actions structure to comply with DMTF Redfish specification
- Removed allowable reset types from action reset

sudhir-intc committed:
fix: got back the facilyt name and room
